src/visualization/components/Snapshot.tsx

```diff
@@ -46,19 +55,45 @@ class Snapshot extends React.Component<any, any> {
         // Even if it's not a different impression, refresh the viewport regardless
         this.parsers["Viewport"].setup(this.frame.contentDocument, this.frame, this.props.base);
       }
-      
+    
+
       var startPointer = time < this.currentTime ? 0 : this.currentPointer + 1;
       for (var i = startPointer; i < events.length; i++) {
         var event = events[i];
         if (event.time <= time) {
-          let parser = event.type === "Layout" && this.props.view == 1 ? "BoxModel" : event.type; 
+          
+          //let parser = event.type === "Layout" && this.props.view == 1 ? "BoxModel" : event.type;
+          let parser = event.type;
+          if(event.type === "Layout"){
+            if(this.props.view == 1){
+              parser = "BoxModel";
+            }else if(this.props.view == 2){
+              parser = "SettleEvents";
+            }else{
+              parser = event.type;
+            }
```

Comment:
your else is misplaced here - you want to default to event.type in cases where type was not layout as well. It would be cleanest to not nest your if statements